core-web/libs/portlets/dot-ema/src/lib/edit-ema-editor/edit-ema-editor.component.html

```diff
@@ -1,70 +1,86 @@
+                    (cdkCopyToClipboardCopied)="triggerCopyToast()"
+                    icon="pi pi-copy"
+                    styleClass="p-button-text p-button-sm"
+                    data-testId="ema-copy-url" />
+                <a
+                    class="p-button-sm p-button-text"
+                    [href]="es.apiURL"
+                    [title]="es.apiURL"
+                    pButton
+                    target="_blank"
+                    data-testId="ema-api-link"
+                    icon="pi pi-link"
+                    label="API">
+                </a>
+            </ng-container>
+            <ng-container right>
+                <p *ngIf="!es.editor.page.canEdit">You don't have permission to edit this page.</p>
+                <dot-edit-ema-language-selector
+                    [language]="es.editor.viewAs.language"
+                    (selected)="onLanguageSelected($event)"
+                    data-testId="language-selector" />
+                <dot-edit-ema-persona-selector
+                    #personaSelector
+                    [pageId]="es.editor.page.identifier"
+                    [value]="es.editor.viewAs.persona"
+                    (selected)="onPersonaSelected($event)"
+                    (despersonalize)="onDespersonalize($event)"
+                    data-testId="persona-selector" />
+                <p-button class="p-button-sm" label="Save" />
+            </ng-container>
+        </dot-edit-ema-toolbar>
+        <div
+            class="iframe-wrapper"
+            [ngStyle]="{ 'grid-column': es.editor.page.canEdit ? '' : 'span 2' }">
+            <iframe
+                #iframe
+                [src]="es.iframeURL | safeUrl"
+                [title]="host"
+                data-testId="iframe"
+                style="border: none"
+                width="100%"
+                height="100%"></iframe>
+            <dot-ema-contentlet-tools
+                *ngIf="!rows.length && !!contentlet && es.editor.page.canEdit"
+                [contentlet]="contentlet"
+                (edit)="editContentlet($event)"
+                (delete)="deleteContentlet($event)"
+                (addWidget)="addWidget($event)"
+                (addForm)="addForm($event)"
+                (addContent)="addContentlet($event)" />
+            <dot-ema-page-dropzone
+                *ngIf="!!rows.length"
+                [rows]="rows"
+                (place)="onPlaceItem($event)"
+                data-testId="dropzone" />
+        </div>
+
+        <dot-edit-ema-palette
+            *ngIf="es.editor.page.canEdit"
+            [languageId]="es.editor.viewAs.language.id"
+            [containers]="es.editor.containers"
+            (dragStart)="onDragStart($event)"
+            (dragEnd)="onDragEnd($event)"></dot-edit-ema-palette>
+    </ng-container>
+
+    <ng-template #cantRead>
+        <p data-testId="dont-have-permission">You don't have permission to read this page.</p>
```

Comment:
Should we use `DotMessagePipe` here to practice internationalization (i18n)?

Reply:
Yep! This is only a temp placeholder until we have the final design